pid:
  position:
    x:
      Kp: 0.1
      Ki: 0
      Kd: 0
      Ff: 0
      control_effort:
        min: -1
        max: 1
      derivative_type: 1
      error_ramp_rate: 1000
    y:
      Kp: 0.1
      Ki: 0
      Kd: 0
      Ff: 0
      control_effort:
        min: -1
        max: 1
      derivative_type: 1
      error_ramp_rate: 1000
    z:
      Kp: 0.1
      Ki: 0
      Kd: 0
      Ff: 0
      control_effort:
        min: -1
        max: 1
      derivative_type: 1
      error_ramp_rate: 1000
    roll:
      Kp: 0.1
      Ki: 0
      Kd: 0
      Ff: 0
      control_effort:
        min: -1
        max: 1
      derivative_type: 1
      error_ramp_rate: 1000
    pitch:
      Kp: 0
      Ki: 0
      Kd: 0
      Ff: 0
      control_effort:
        min: 0
        max: 0
      derivative_type: 1
      error_ramp_rate: 1000
    yaw:
      Kp: 0.1
      Ki: 0
      Kd: 0
      Ff: 0
      control_effort:
        min: -1
        max: 1
      derivative_type: 1
      error_ramp_rate: 1000
  position_cascaded:
    x:
      Kp: 4
      Ki: 0
      Kd: 5
      Ff: 0
      control_effort:
        min: -1
        max: 1
      derivative_type: 1
      error_ramp_rate: 1000
    y:
      Kp: 3.5
      Ki: 0
      Kd: 5
      Ff: 0
      control_effort:
        min: -1
        max: 1
      derivative_type: 1
      error_ramp_rate: 1000
    z:
      Kp: 1
      Ki: 0
      Kd: 1.5
      Ff: 0
      control_effort:
        min: -1
        max: 1
      derivative_type: 1
      error_ramp_rate: 1000
    roll:
      Kp: 0.8
      Ki: 0
      Kd: 0.4
      Ff: 0
      control_effort:
        min: -1
        max: 1
      derivative_type: 1
      error_ramp_rate: 1000
    pitch:
      Kp: 0
      Ki: 0
      Kd: 0
      Ff: 0
      control_effort:
        min: 0
        max: 0
      derivative_type: 1
      error_ramp_rate: 1000
    yaw:
      Kp: 1
      Ki: 0
      Kd: 0.5
      Ff: 0
      control_effort:
        min: -1
        max: 1
      derivative_type: 1
      error_ramp_rate: 1000
  velocity:
    x:
      Kp: 1.75
      Ki: 0.02
      Kd: 0.5
      Ff: 0
      control_effort:
        min: -1
        max: 1
      derivative_type: 0
      error_ramp_rate: 1000
    y:
      Kp: 1
      Ki: 0.02
      Kd: 0.05
      Ff: 0
      control_effort:
        min: -1
        max: 1
      derivative_type: 0
      error_ramp_rate: 1000
    z:
      Kp: 0.5
      Ki: 0.5
      Kd: 0.5
      Ff: 0
      control_effort:
        min: -1
        max: 1
      derivative_type: 0
      error_ramp_rate: 1000
    roll:
      Kp: 0.01
      Ki: 0.001
      Kd: 0.01
      Ff: 0
      control_effort:
        min: -1
        max: 1
      derivative_type: 0
      error_ramp_rate: 1000
    pitch:
      Kp: 0
      Ki: 0
      Kd: 0
      Ff: 0
      control_effort:
        min: 0
        max: 0
      derivative_type: 0
      error_ramp_rate: 1000
    yaw:
      Kp: 0.01
      Ki: 0.001
      Kd: 0.01
      Ff: 0
      control_effort:
        min: -1
        max: 1
      derivative_type: 0
      error_ramp_rate: 1000
desired_power_limits:
  x:
    min: -1
    max: 1
  y:
    min: -1
    max: 1
  z:
    min: -1
    max: 1
  roll:
    min: -1
    max: 1
  pitch:
    min: -1
    max: 1
  yaw:
    min: -1
    max: 1
static_power_global:
  x: 0
  y: 0
  z: 0
power_scale_factor: 1
thrusters:
  - name: middle_right
    type: T200
    pos: [-0.2404152701, -0.42425, -0.003175]
    rpy: [0, 90, 0]
    flipped: true
  - name: front_right
    type: T200
    pos: [3.360246e-05, -0.3867988726, -0.003175]
    rpy: [0, 0, -135]
    flipped: false
  - name: back_right
    type: T200
    pos: [-0.4808641427, -0.3867988726, -0.003175]
    rpy: [0, 0, 135]
    flipped: true
  - name: back_left
    type: T200
    pos: [-0.4808641427, 0.09409887258, -0.003175]
    rpy: [0, 0, -135]
    flipped: false
  - name: middle_left
    type: T200
    pos: [-0.2404152701, 0.13155, -0.003175]
    rpy: [0, 90, 0]
    flipped: true
  - name: front_left
    type: T200
    pos: [3.360246e-05, 0.09409887258, -0.003175]
    rpy: [0, 0, 135]
    flipped: true
wrench_matrix_file_path: data/crush_wrench.csv
wrench_matrix_pinv_file_path: data/crush_wrench_pinv.csv